components/boards/AllArticles.tsx

```diff
@@ -0,0 +1,134 @@
+import axios from "@/lib/axios";
+import Image from "next/image";
+import { Article, ArticlesResponse } from "@/types/types";
+import React, { ChangeEvent, useEffect, useState } from "react";
+import profileIcon from "@/public/images/icons/ic_profile.svg";
+import heartIcon from "@/public/images/icons/ic_heart.svg";
+import searchIcon from "@/public/images/icons/ic_search.svg";
+
+import styles from "./AllArticles.module.css";
+import { useRouter } from "next/router";
+import Link from "next/link";
+
+const AllArticles = () => {
+  const [articles, setArticles] = useState<Article[]>([]);
+  const [order, setOrder] = useState<string>("recent");
+  const [searchKeyword, setSearchKeyword] = useState<string>("");
+
+  const router = useRouter();
+  const keyword = (router.query.keyword as string) || "";
+
+  useEffect(() => {
+    const getAllArticles = async () => {
+      let params = `/articles?page=1&pageSize=10&orderBy=${order}`;
+      if (keyword.trim()) {
+        params += `&keyword=${encodeURIComponent(keyword)}`;
+      }
+      const res = await axios.get<ArticlesResponse>(params);
+      const articles = res.data;
+      setArticles(articles.list);
+    };
+
+    getAllArticles();
+  }, [order, keyword]);
+
+  useEffect(() => {
+    setSearchKeyword(keyword);
+  }, [keyword]);
+
+  const dateFormat = (date: Date) => {
+    const newDate = new Date(date);
+    const formatDate = `${newDate.getFullYear()}.${String(
+      newDate.getMonth() + 1
+    ).padStart(2, "0")}.${String(newDate.getDate()).padStart(2, "0")}`;
+    return formatDate;
+  };
+
+  const handleOrderChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    setOrder(e.target.value);
+  };
+
+  const handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSearchKeyword(e.target.value);
+  };
+  const handleSearchKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const query = { ...router.query };
+    if (e.key === "Enter") {
+      if (searchKeyword.trim()) {
+        query.keyword = searchKeyword;
+      } else {
+        delete query.keyword;
+      }
+
+      router.replace({
+        pathname: router.pathname,
+        query: query,
+      });
+    }
+  };
```

Comment:
```suggestion
const updateRouterQuery = (query)=>{
      router.replace({
        pathname: router.pathname,
        query: query,
      });
}

  const handleSearchKeyword = (e: React.KeyboardEvent<HTMLInputElement>) => {
    const query = { ...router.query };
    if (e.key === "Enter") {
      if (searchKeyword.trim()) {
        query.keyword = searchKeyword;
      } else {
        delete query.keyword;
      }
    }
    updateRouterQuery(query)
  };
  
```


* 함수 하나에 2가지 일을 하고 있어서 2개로 분리하는건 어떨까요?
* 분리하면 검색어를 업데이트 하는 함수, 그리고 라우터 쿼리를 업데이트 하는 함수 이렇게 2개로 나눌 수 있을것같아요.